src/infrastructure/api/index.ts

```diff
@@ -20,111 +15,24 @@ app.get('/version', (_request, res) => {
-          // You will need to create your s3local profile (~/.aws/credentials) if you are not using any
-          accessKeyId: 'S3RVER',
-          secretAccessKey: 'S3RVER',
-          endpoint: new AWS.Endpoint('http://localhost:4569'),
-        },
-      ),
-      `cvs-cert-${BUCKET}`,
-      BRANCH,
-      NODE_ENV,
-    )
-      .then(({ statusCode, headers, body }) => {
-        res.status(statusCode);
-
-        if (headers) {
-          res.header(headers);
-        }
-
-        res.send(body);
-      })
-      .catch((e: Error) => {
-        console.error(e.message);
-        res.status(500).send(e.message);
-      });
-  } else if (req.query.plateSerialNumber && !req.query.vinNumber && !req.query.testNumber && !req.query.systemNumber) {
-    console.info('Calling plate service');
-
-    getPlate(
-      {
-        plateSerialNumber: req.query.plateSerialNumber as string,
-      },
-      new S3(
-        process.env.IS_OFFLINE && {
-          s3ForcePathStyle: true,
-          // You will need to create your s3local profile (~/.aws/credentials) if you are not using any
-          accessKeyId: 'S3RVER',
-          secretAccessKey: 'S3RVER',
-          endpoint: new AWS.Endpoint('http://localhost:4569'),
-        },
-      ),
-      `cvs-cert-${BUCKET}`,
-      BRANCH,
-      NODE_ENV,
-    )
-      .then(({ statusCode, headers, body }) => {
-        res.status(statusCode);
-
-        if (headers) {
-          res.header(headers);
-        }
-
-        res.send(body);
-      })
-      .catch((e: Error) => {
-        console.error(e.message);
-        res.status(500).send(e.message);
-      });
-  } else if (!req.query.plateSerialNumber && req.query.vinNumber && !req.query.testNumber && req.query.systemNumber) {
-    console.info('Calling letter service');
-
-    
-    getLetter(
-      {
-        vin: req.query.vinNumber as string,
-        systemNumber: req.query.systemNumber as string,
-      },
-      new S3(
-        process.env.IS_OFFLINE && {
-          s3ForcePathStyle: true,
-          // You will need to create your s3local profile (~/.aws/credentials) if you are not using any
-          accessKeyId: 'S3RVER',
-          secretAccessKey: 'S3RVER',
-          endpoint: new AWS.Endpoint('http://localhost:4569'),
-        },
-      ),
-      `cvs-cert-${BUCKET}`,
-      BRANCH,
-      NODE_ENV,
-    )
-      .then(({ statusCode, headers, body }) => {
-        res.status(statusCode);
-
-        if (headers) {
-          res.header(headers);
-        }
-
-        res.send(body);
-      })
-      .catch((e: Error) => {
-        console.error(e.message);
-        res.status(500).send(e.message);
-      });
-  } else {
-    res.status(400).end();
-  }
+  const {
+    vinNumber, plateSerialNumber, testNumber, systemNumber,
+  } = req.query;
+
+  documentRequestFactory(vinNumber as string, testNumber as string, plateSerialNumber as string, systemNumber as string)
+    .then(({ statusCode, headers, body }) => {
```

Comment:
I'd be in favour of having this to be an await, given that the logic happening inside isn't a lot

Reply:
I would prefer to use await, but the express function isn't async so we can't